website/docs/reference/resource-configs/postgres-configs.md

```diff
@@ -108,21 +108,43 @@ models:
 
 ## Materialized views
 
-The Postgres adapter supports [materialized views](https://www.postgresql.org/docs/current/rules-materializedviews.html).
-Indexes are the only configuration that is specific to `dbt-postgres`.
-The remaining configuration follows the general [materialized view](/docs/build/materializations#materialized-view) configuration.
-There are also some limitations that we hope to address in the next version.
+The Postgres adapter supports [materialized views](https://www.postgresql.org/docs/current/rules-materializedviews.html)
+with the following configuration parameters:
 
-### Monitored configuration changes
+| Parameter                 | Type       | Required | Default   | Change Monitoring Support | Reference           |
+|---------------------------|------------|----------|-----------|---------------------------|---------------------|
+| `on_configuration_change` | STRING     | NO       | `'apply'` | N/A                       |                     |
+| `indexes`                 | LIST[DICT] | NO       |           | ALTER                     | [Indexes](#indexes) |
 
-The settings below are monitored for changes applicable to `on_configuration_change`.
+#### Sample model file:
 
-#### Indexes
+<File name='postgres_materialized_view.sql'>
 
-Index changes (`CREATE`, `DROP`) can be applied without the need to rebuild the materialized view.
-This differs from a table model, where the table needs to be dropped and re-created to update the indexes.
-If the `indexes` portion of the `config` block is updated, the changes will be detected and applied
-directly to the materialized view in place.
+```sql
+{{ config(
+    materialized='materialized_view',
+    on_configuration_change='{ apply | continue | fail }'
+    indexes=[
+        {
+            'columns': ['<column_name>', ...],
+            'unique': <bool>},
+            'type': '{ HASH | B-TREE | GIST | SP-GIST | GIN | BRIN }',
+    ]
+) }}
+
+select * from {{ ref('my_base_table') }}
+```
+
+</File>
+
+The `indexes` parameter corresponds to that of a table, as linked above.
+It's worth noting that, unlike with tables, dbt will monitor this parameter for changes and apply the changes without dropping the materialized view.
+This happens via a `DROP/CREATE` of the indexes, which could be thought of as a `ALTER` of the materialized view.
+
+Find more information about materialized view parameters in the Postgres docs:
+- [CREATE MATERIALIZED VIEW](https://www.postgresql.org/docs/current/sql-creatematerializedview.html)
+
+<VersionBlock lastVersion="1.6">
```

Comment:
I'm actually not sure how nested version blocks work. If they stack, I think we should only specify `lastVersion` since `firstVersion` is already specified in the outer block. Clearly I went back and forth on this since I wound up submitting both versions.

Reply:
After thinking about this, I would rather exclude `firstVersion`. Keeping just `lastVersion` aligns with the idea that this limitation was removed in 1.7. Adding `firstVersion` seems to suggest this limitation was introduced in 1.6, which is accurate, but misleading.